lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,21 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
```

Comment:
Good idea! I've done a first pass at using the parser gem. That what you had in mind @chastell ?